ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -353,6 +368,43 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
           subItem->setMaterial(this->heatSignatureMaterials[item->getId()]);
         }
       }
+      // background objects
+      else
+      {
+        Ogre::Aabb aabb = item->getWorldAabbUpdated();
+        Ogre::AxisAlignedBox box = Ogre::AxisAlignedBox(aabb.getMinimum(), aabb.getMaximum());
+
+        // we will be converting rgb values to tempearture values in shaders
+        // but we want to make sure the object rgb values are not affected by lighting
+        // so disable lighting
+        // Also check if objects are within camera view
+        if (ogreVisual->GeometryCount() > 0u && this->ogreCamera->isVisible(box))
+        {
+          auto geom = ogreVisual->GeometryByIndex(0);
+          MaterialPtr mat = geom->Material();
+          Ogre2MaterialPtr ogreMat = std::dynamic_pointer_cast<Ogre2Material>(mat);
+          Ogre::HlmsUnlitDatablock *unlit = ogreMat->UnlitDatablock();
+
+          for (unsigned int i = 0; i < item->getNumSubItems(); ++i)
+          {
+            Ogre::SubItem *subItem = item->getSubItem(i);
+            auto it = this->unlitDatablockMap.find(subItem);
+            if (it == this->unlitDatablockMap.end())
+            {
+              this->unlitDatablockMap[subItem] = unlit;
```

Comment:
Can you help me understand the purpose of this assignment? I don't see `unlitDatablockMap` being accessed anywhere else. Is `unlitDatablockMap` by chance a testing/outdated variable that can be removed?

Reply:
you're right, we don't need to store it. I'll remove it.

Reply:
removed. 397a443